src/ansible_creator/config.py

```diff
@@ -23,14 +24,31 @@ class Config:
     collection: str = ""
     force: bool = False
     init_path: str = "./"
+    project: str = ""
+    scm_org: str = ""
+    scm_project: str = ""
 
     # TO-DO: Add instance variables for other 'create' and 'sample'
 
-    collection_name: str = field(init=False)
-    namespace: str = field(init=False)
+    collection_name: str = ""
+    namespace: str = ""
 
     def __post_init__(self: Config) -> None:
         """Post process config values."""
+        # Show CreatorError if the required collection name is not provided
+        if not self.collection and self.project == "collection":
+            msg = "The collection name is required when scaffolding a collection."
+            raise CreatorError(msg)
+
+        if self.project == "ansible-project" and (
+            self.scm_org is None or self.scm_project is None
+        ):
+            msg = (
+                "Required parameters scm-org and scm-project to scaffold"
```

Comment:
Show a warning message when --scm-org and --scm-project is specified with 
--project=collection.

Use `output.warning`

Reply:
This should happen in cli.py^